config.go

```diff
@@ -114,6 +114,24 @@ func (c *adConfig) UpdateAdRequest(req *AdRequest) {
 
 	if len(req.DisplayAreas) == 0 {
 		req.DisplayAreas = c.baseRequest.DisplayAreas
+	} else {
+		mergedDisplayArea := req.DisplayAreas[0]
+		paramDisplayArea := c.baseRequest.DisplayAreas[0]
+
+		if mergedDisplayArea.Width == 0 {
+			mergedDisplayArea.Width = paramDisplayArea.Width
+		}
+
+		if mergedDisplayArea.Height == 0 {
+			mergedDisplayArea.Height = paramDisplayArea.Height
+		}
+
+		if mergedDisplayArea.SupportedMedia == nil {
+			mergedDisplayArea.SupportedMedia = paramDisplayArea.SupportedMedia
```

Comment:
I'll update this line length issue in the next commit, this is the first pass, have few questions, will ask them tomorrow in the office.